### PR TITLE
Analyzing effect of dimensionality reduction on accuracy of different classifiers on different types of datasets

### PR DESCRIPTION
**Pull request by parimal173**
It will be a document showing the Effect of Dimensionality Reduction on the accuracy of different classifiers.
The document will have simulations On High dimensional dataset of different shapes:
Each dataset is synthesized from sklearn established Datasets. Each dataset has 1000 dimensions with only 2 dimensions of data and rest are noise dimensions.

Questions to answer:

Analyzing how dimensionality reduction helps in classification for different classifiers.
Analyzing how classifiers perform with a different number of reduced datasets from the main high dimensional dataset.
Pipeline to be followed:

Defining a dataset with sklearn established synthetic datasets with High dimensions.
Performing classification on data and measuring accuracy for quantification of the process.
performing the Dimensionality reduction technique keeping varying numbers of reduced dimensions.
Checking the performance of classification again after reducing dimension after each iteration.
The output of the PR would be a figure showing different datasets, comparing accuracies of different classifiers with and without dimensionality reduction and a plot showing varying accuracies over reduced dimensions.

Experiments to follow:
https://github.com/parimal173/scikit-learn/blob/parimal173-patch-1/Final_50trials_9datasets.ipynb

### CODE REVIEW

**Comment by bdpedigo** [comment]
- `Xp=dataset
Yp=labels` is unnecessary 
- in cell run [4] title the datasets and remove axes ticks and ticklabels
- the main block of your code, cell run [10] is extremely hard code to read because of how many lists and things you have going on, and It is impossible to read stuff like `dsvm1 += dim_data1[(198 * (i+1)):(297 * (i+1))]` and know what is going on. Never hard code numbers like this. You should be able to do all of this a lot more succinctly with dataframes, or, look at how other people on your team have structured similar simulations. 
- 2 simulations is not enough, try 50? if that takes too long, then only do like 5 values of n_dimensions (probably on a log scale
- all plots need axes labels... 
- if the legend is getting in the way of the figure, move it outside of the figure
- I would not flip the xaxis order as you have done in the bottom figure. I think this is the only figure we need (can remove the others along the way). remove all of the legends besides the last one. 

I have a hard time interpreting results mainly because I cannot read that big code block. I don't think I trust results - for example, how is it possible that after adding 98 dimensions of pure noise, these classifiers would do better with the noisy data than just the two signal dimensions? Am I interpreting that correctly?

**Comment by parimal173** [comment]
> * `Xp=dataset Yp=labels` is unnecessary
> * in cell run [4] title the datasets and remove axes ticks and ticklabels
> * the main block of your code, cell run [10] is extremely hard code to read because of how many lists and things you have going on, and It is impossible to read stuff like `dsvm1 += dim_data1[(198 * (i+1)):(297 * (i+1))]` and know what is going on. Never hard code numbers like this. You should be able to do all of this a lot more succinctly with dataframes, or, look at how other people on your team have structured similar simulations.
> * 2 simulations is not enough, try 50? if that takes too long, then only do like 5 values of n_dimensions (probably on a log scale
> * all plots need axes labels...
> * if the legend is getting in the way of the figure, move it outside of the figure
> * I would not flip the xaxis order as you have done in the bottom figure. I think this is the only figure we need (can remove the others along the way). remove all of the legends besides the last one.
> 
> I have a hard time interpreting results mainly because I cannot read that big code block. I don't think I trust results - for example, how is it possible that after adding 98 dimensions of pure noise, these classifiers would do better with the noisy data than just the two signal dimensions? Am I interpreting that correctly?

I have made chages. About the results, i tred it multiple times and I am getting same results.

**Comment by bdpedigo** [comment]
you are scaling the data after adding the noise so each row is the same experiment

**Comment by bdpedigo** [comment]
> in cell run [4] title the datasets and remove axes ticks and ticklabels

address this

**Comment by bdpedigo** [comment]
> all plots need axes labels...
address this

**Comment by parimal173** [comment]
> you are scaling the data after adding the noise so each row is the same experiment

I have made changes and added PR to NeuroDataDesign repo. Please tell me if its correct.
https://github.com/NeuroDataDesign/scikit-learn/pull/23
